status/reporters.go

```diff
@@ -679,13 +690,24 @@ func (r *GitHubReporter) ReportStatusForSnapshot(k8sClient client.Client, ctx co
 
 		for _, integrationTestStatusDetail := range integrationTestStatusDetails {
 			integrationTestStatusDetail := *integrationTestStatusDetail //G601
+
+			pipelineRun, err := helpers.GetPipelineWithName(integrationTestStatusDetail.TestPipelineRunName, snapshot.Namespace, k8sClient, ctx)
+			if err != nil {
+				return err
+			}
+
 			commitStatus, err := r.createCommitStatusAdapterForSnapshot(snapshot, integrationTestStatusDetail, owner, repo, sha)
 			if err != nil {
 				logger.Error(err, "failed to create CommitStatusAdapter for snapshot",
 					"snapshot.NameSpace", snapshot.Namespace, "snapshot.Name", snapshot.Name)
 				return err
 			}
 
+			err = helpers.RemoveFinalizer(k8sClient, *logger, ctx, pipelineRun)
```

Comment:
I don't think that finalizer should be removed here
This function may fail later and we would still need the pipeline details on requeue.

also from SRP, this function shouldn't care about finalizers, it shold only care about updating status in GH. I think removal of finalizers should be done as separate function, once final GH report has been done.